### ISSUE TITLE
Add lint to find uses of end-of-line comments where doc comments were likely intended

### ISSUE THREAD

**Issue by bwilkerson** [dart-lang/linter#727]
It would be nice to have a lint to catch cases like the following
```
// method comment
void m() {}
```
There might be too many false positives, although we can special case some comments, such as the comments to ignore lints.

**Comment by zoechi** [comment]
It's probably redundant when http://dart-lang.github.io/linter/lints/package_api_docs.html is used

**Comment by bwilkerson** [comment]
Good point, but we provide doc comments on private methods when they are complex enough to warrant it, and I've seen the same mistake made there.

**Comment by zoechi** [comment]
I sometimes add comments that are intentionally not doc comments. 
There should be an easy way to opt out to avoid false positives.
(I don't need to enable this lint of course, just trying to imagine how it would affect me if I did)
